scripts/eurostat/health_determinants/common/EuroStat.py

```diff
@@ -0,0 +1,359 @@
+            split_columns = split_columns.replace('frequenc',
+                                                  'frequenc_alcohol')
+            df.rename(columns={
+                'frequenc': 'frequenc_alcohol',
+            }, inplace=True)
+        elif import_name == "tobacco_consumption":
+            split_columns = split_columns.replace('frequenc',
+                                                  'frequenc_tobacco').replace(
+                                                      'quantile', 'quant_inc')
+            df.rename(columns={
+                'frequenc': 'frequenc_tobacco',
+                'quantile': 'quant_inc'
+            },
+                      inplace=True)
+        elif import_name == "bmi":
+            if 'quant_inc' in df.columns.values.tolist():
+                df = df[(~(df['quant_inc'] == 'UNK'))]
+
+        df = df[df['age'] == 'TOTAL']
+
+        for col in [
+                'sex', 'quant_inc', 'frequenc', 'frequenc_alcohol',
+                'frequenc_tobacco', 'isced11', 'deg_urb', 'bmi', 'duration',
+                'c_birth', 'citizen', 'smoking', 'duration', 'physact',
+                'lev_limit', 'levels'
+        ]:
+            if col in df.columns.values.tolist():
+                df = _replace_col_values(df, col)
+
+        df['SV'] = eval(file_to_sv_mapping[import_name][file_name])
+
+        split_columns_list = split_columns.split(',')
+        if 'time' in split_columns_list:
+            split_columns_list.remove('time')
+            id_vars = ['SV', 'time']
+            var_name = 'geo'
+        if 'geo' in split_columns_list:
+            split_columns_list.remove('geo')
+            id_vars = ['SV', 'geo']
+            var_name = 'time'
+
+        df.drop(columns=split_columns_list, inplace=True)
+        df = df.melt(id_vars=id_vars,
+                     var_name=var_name,
+                     value_name='observation')
+        df = df[df['geo'] != 'EU27_2020']
+        df = df[df['geo'] != 'EU28']
+        return df
+
+    def process(self) -> list:
+        """
+        This Method calls the required methods to generate
+        cleaned CSV, MCF, and TMCF file.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        final_df = pd.DataFrame(
+            columns=['time', 'geo', 'SV', 'observation', 'Measurement_Method'])
+        # Creating Output Directory
+        output_path = os.path.dirname(self._cleaned_csv_file_path)
+        if not os.path.exists(output_path):
+            os.mkdir(output_path)
+        sv_list = []
+
+        for file_path in self._input_files:
+            df = pd.read_csv(file_path, sep='\t', header=0)
+            file_name = file_path.split("/")[-1][:-4]
+            df.columns = df.columns.str.strip()
+            df = self._parse_file(file_name, df, self._import_name)
+            df['SV'] = df['SV'].str.replace('_Total', '')
+            final_df = pd.concat([final_df, df])
```

Comment:
instead of concating two at a time, create a list of df to be concated and merge in a call outside the loop